lib/android_build/maesdk/src/main/java/com/microsoft/applications/events/SubstrateSignals.java

```diff
@@ -0,0 +1,29 @@
+package com.microsoft.applications.events;
+
+public class SubstrateSignals {
+    /**
+     * Creates EventProperties object from json SignalItem.
+     *
+     * @param loggerPtr the ILogger pointer
+     * @param signalItemJson the Signal Item in json format.
+     */
+    public static native boolean sendSignal(long loggerPtr, String signalItemJson);
```

Comment:
Just curious, how is this method going to be invoked ?

Reply:
Hey, it will be called after an user interaction happened. The first candidate would be Microsoft Teams "LinkClicked" signal, that is generated by the client when the user clicks on a link.